backend/src/bundles/users/user.repository.ts

```diff
@@ -32,16 +33,32 @@ class UserRepository implements Repository {
 
     public async create(entity: UserEntity): Promise<UserEntity> {
         const { email, passwordHash } = entity.toNewObject();
-        const user = await this.userModel
-            .query()
-            .insert({
-                email,
-                passwordHash,
-            })
-            .returning('*')
-            .execute();
+        const trx = await this.userModel.startTransaction();
+
+        try {
+            const user = await this.userModel
+                .query(trx)
+                .insert({
+                    email,
+                    passwordHash,
+                })
+                .returning('*')
+                .execute();
+
+            const userDetails = await user
+                .$relatedQuery('userDetails', trx)
+                .insert({ userId: user.id });
```

Comment:
not sure about this one, but specifying here `userId` should be unnecessary, since query is already related to specific user.

Reply:
Okay, I'll test the creation of user without passing userId explicitly